spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -1819,20 +1827,20 @@ def _do_load(self, application_graph_changed):
         optional_algorithms.append("WriteMemoryIOData")
 
         if self._exec_dse_on_host:
-            optional_algorithms.append("HostExecuteDataSpecification")
+            optional_algorithms.append("HostExecuteOtherDataSpecification")
         else:
-            optional_algorithms.append("MachineExecuteDataSpecification")
+            optional_algorithms.append("MachineExecuteOtherDataSpecification")
 
         # Reload any parameters over the loaded data if we have already
         # run and not using a virtual board
         if self._has_ran and not self._use_virtual_board:
             optional_algorithms.append("DSGRegionReloader")
 
-        # Get the executable targets
-        optional_algorithms.append("GraphBinaryGatherer")
-
         # algorithms needed for loading the binaries to the SpiNNaker machine
-        optional_algorithms.append("LoadExecutableImages")
+        optional_algorithms.append("LoadOtherExecutableImages")
+        if enable_advanced_monitor:
```

Comment:
This is wrong. It will make things blow up if users turn on reinjection but don't turn on advanced monitors. It is simpler (by far!) to just apply the algorithms always, but ensure that they do nothing gracefully when there's nothing to do.